From 345e89f1ddd0d8a5a84ee4cedc57adaa9911c84b Mon Sep 17 00:00:00 2001
From: Thomas Guillem <thomas@gllm.fr>
Date: Wed, 20 Apr 2022 17:02:50 +0200
Subject: [PATCH 77/80] smb2: fix anonymous login

Not sure when this regression happened on the libsmb2 side.
But setting the password to an empty string do enable anonymous login
now.
---
 modules/access/smb2.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/access/smb2.c b/modules/access/smb2.c
index 8d0f3980ea..ff41707a63 100644
--- a/modules/access/smb2.c
+++ b/modules/access/smb2.c
@@ -647,8 +647,8 @@ vlc_smb2_connect_open_share(stream_t *access, const char *url,
     if (!username)
     {
         username = "Guest";
-        /* A NULL password enable ntlmssp anonymous login */
-        password = NULL;
+        /* An empty password enable ntlmssp anonymous login */
+        password = "";
     }
 
     struct vlc_access_cache_entry *cache_entry =
-- 
2.32.0 (Apple Git-132)

